RELEASE.md

```diff
@@ -0,0 +1,74 @@
+# Versioning and releasing Google Api Compiler
+This document states how we plan to version and release the Google Api Compiler.
+1. How and when to release Google Api Compiler
+2. How to setup development branch for day to day commits.
+
+## Branches
+This Api Compiler repo contains only _one_ branch:
+
+`master` : This has the lastest code into which all the contribution
+(pull requests) goes. From this branch we will create a temporary release branch
+through which we will do the release to the central repository. After the
+release the temporary release branch will be deleted.
+
+## Steps to release a new version
+1. Once every month (ideally last Monday of every month) move the code from
+master branch into the release branch (create it in the same repo).
+
+2. Now in the release branch bump up the version number in the build.gradle by
+   running
+   ```
+   utils/update_gradle_version.sh
+   ```
+3. Now update the README.md to match the new version.
+   ```
+   utils/update_doc_version.sh
+   ```
+4. Now create a PR with the changes to build.gradle and README.
+5. Once your PR is merged in the release branch, ensure that it builds without
+errors on the [travis PR build status page](https://travis-ci.org/googleapis/api-compiler/pull_requests)
+
+6. Now push the jars to central repository.
+
+   Step1: On you local machine create a file ~/.gradle/gradle.properties.
+   Edit the file with the following content:
+   ```
+   # TODO: Add link to the document on how to get these Key and password.
+   signing.keyId=YOUR-KEY-ID-HERE
+   signing.password=YOUR-PASSWORD-HERE
+   signing.secretKeyRingFile=/usr/local/google/home/YOUR-USER-NAME/.gnupg/secring.gpg
+
+   privateOssrhUsername=<username that has access to the central repository>
+   privateOssrhPassword=<password associated with the above username>
+   ```
+   
+   Step2: Push the changes to the central repository using after_success.sh.
+   This script uses the information in the  ~/.gradle/gradle.properties to
+   successfully deploy jars to the central repository.
+   ```
+   # Run the command from the release branch.
+   utils/after_success.sh
+   ```
+7. After successful push to central repository, create a label for that release
+branch and then delete the branch.
+
+## Setup master branch for development of the next release.
+1. Now in the master branch pump up the version number _twice_ in the
```

Comment:
Its an extra step. You can directly set the version you want as  parameter. 

The current default code has:
1. Go from x.y.z-SNAPSHOT to x.y.z, because that is the logical next version.
2. Also  x.y.z to x.y.z+1-SNAPSHOT, because that is the next logical version.

However, the workflow I have proposed here is to move the master by two versions directly and therefore need to run the script twice. There can be script changes to avoid running twice, but I think if our workflow changes in the future (highly likely) this existing script will work without any changes.